src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -55,14 +115,55 @@ internal ConfigurationClientManager(IList<ConfigurationClientWrapper> clients)
             _clients = clients;
         }
 
-        public IEnumerable<ConfigurationClient> GetAvailableClients(DateTimeOffset time)
+        public IEnumerable<ConfigurationClient> GetAvailableClients()
         {
-            return _clients.Where(client => client.BackoffEndTime <= time).Select(c => c.Client).ToList();
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+
+            if (_replicaDiscoveryEnabled && IsFallbackClientDiscoveryDue(now))
+            {
+                _lastFallbackClientRefreshAttempt = now;
+
+                _ = DiscoverFallbackClients();
+            }
+
+            List<ConfigurationClient> clients = new List<ConfigurationClient>(_clients.Where(c => c.BackoffEndTime <= now).Select(c => c.Client));
```

Comment:
why copy?

Reply:
Actually, we don't have to worry about this, updated.